.github/workflows/withdraw-packages.yaml

```diff
@@ -0,0 +1,50 @@
+name: Withdraw packages
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Withdraw packages
+    runs-on: ubuntu-latest
+
+    permissions:
+      packages: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: chainguard-dev/actions/setup-melange@main
+
+      - uses: google-github-actions/setup-gcloud@v0
+
+      - run: echo "${{ secrets.MELANGE_RSA }}" > ./wolfi-signing.rsa
+      - run: |
+          sudo mkdir -p /etc/apk/keys
+          sudo cp ./wolfi-signing.rsa.pub /etc/apk/keys/wolfi-signing.rsa.pub
+
+      - name: 'Delete withdrawn packages'
+        run: |
+          for arch in x86_64 aarch64; do
+            for pkg in $(cat withdrawn-packages.txt); do
+              echo "=> $pkg"
+              gsutil -m rm gs://wolfi-production-registry-destination/os/$arch/$pkg
+            done
+          done
+
+      - name: 'Sync public package repository'
+        run: |
+          mkdir "${{ github.workspace }}/packages"
+          gsutil -m rsync -r gs://wolfi-production-registry-destination/os/ "${{ github.workspace }}/packages/"
```

Comment:
This copies a lot of bytes out of the repo when we really just need to scan the first few KB and get the total object size, which will make this a lot faster. Did you try `wolfictl generate-index`? It should work (🤞) and works against remote objects in GCS.

Reply:
It ran out of file descriptors while scanning the bucket.  While I agree that it would be better, this way works right now.